igcollect/logfile_values.py

```diff
@@ -146,22 +160,22 @@ def parse_args():
     parser.add_argument('--file', default='/var/log/messages')
     parser.add_argument('--columns-num', default='5', type=int)
     parser.add_argument('--metric', type=Metric, nargs='+')
+    parser.add_argument('--time-column', default='0', type=int)
     parser.add_argument('--time-format', default='%Y-%m-%dT%H:%M:%S%z')
     parser.add_argument('--arch', action='store_true')
     parser.add_argument('--debug', '-d', action='store_true')
     return parser.parse_args()
 
 
-def get_metrics_values(line, metrics, time_format, columns_num):
+def get_metrics_values(line, metrics, time_format, columns_num, time_column):
     fields = line.split()
-    if len(fields) == columns_num:
-        timestamp = convert_to_timestamp(fields[0], time_format)
-        for metric in metrics:
-            if timestamp > metric.now - metric.get_timeshift():
-                value = metric.estimate_columns_value(fields)
-                metric.values.append(value)
-            else:
-                return False
+    timestamp = convert_to_timestamp(fields[time_column], time_format)
```

Comment:
Real question: Is it safe to have no check for a possible KeyError beforehand here?

Reply:
The script just will not work then